htmresearch/algorithms/hierarchical_clustering.py

```diff
@@ -71,25 +98,155 @@ def getLinkageMatrix(self):
     Returns a linkage matrix of the form defined by
     http://docs.scipy.org/doc/scipy-0.16.0/reference/generated/scipy.cluster.hierarchy.linkage.html
     """
+    if self._linkage is None:
+      raise LinkageNotComputedException
```

Comment:
You could also just raise a RuntimeError() with a string explaining the exact error. Much simpler. Not a big deal to keep it the way it is.